packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -45,39 +47,69 @@ export default createConnector({
     const results = getResults(searchResults, this.context);
 
     this._allResults = this._allResults || [];
-    this._previousPage = this._previousPage || 0;
+    this._prevState = this._prevState || '';
 
     if (!results) {
       return {
         hits: [],
+        hasPrevious: false,
         hasMore: false,
+        refine: () => {},
+        refinePrevious: () => {},
+        refineNext: () => {},
       };
     }
 
-    const { hits, hitsPerPage, page, nbPages } = results;
+    const {
+      page,
+      hits,
+      hitsPerPage,
+      nbPages,
+      _state: { page: p, ...currentState } = {},
```

Comment:
I tried to use the `searchState` but I got some issues with it, since it is not totally in sync with the `SearchParameters`.

When I update the query, the `getProvidedProps` method get called 2 times:

#### Initial State
`searchState`
```json
{"page": 3}
```
`SearchParameters`
```json
{"page": 2, "query": ""}
```

#### First call
`searchState`
```json
{"page": 1, "query": "g"}
```
`SearchParameters`
```json
{"page": 2, "query": ""}
```
Additionally `results.hits` still contains the old hits

#### Second call
`searchState`
```json
{"page": 1, "query": "g"}
```
`SearchParameters`
```json
{"page": 0, "query": "g"}
```
Now `results.hits` is containing the new hits.

In my case I want to reset the widget on any query or refinement changes and when I have the new hits to display, so on the second render. But if I make my condition on `searchState` instead of `SearchParameters` then I will reset the widget on the first render.

Another solution I have is to put a flag during the first render to mark the widget as "to be reset"  and to reset it on the second render. It works, but that's assuming that we will always want to reset on the following render, is it the case?

Do you see a better solution to this issue using `searchState`?

Reply:
Ok, I'm fine with this solution.